components/support/nimbus-fml/src/backends/kotlin/gen_structs/structural.rs

```diff
@@ -216,44 +254,39 @@ impl CodeType for ListCodeType {
         )
     }
 
-    /// The language specific expression that gets a value of the `prop` from the `vars` object.
-    fn get_value(&self, oracle: &dyn CodeOracle, vars: &dyn Display, prop: &dyn Display) -> String {
-        let v_type = oracle.find(&self.inner);
-
-        let getter = format!(
-            "{vars}.get{vt}List({prop})",
-            vars = vars,
-            vt = v_type.variables_type(oracle),
-            prop = identifiers::quoted(prop),
-        );
-
-        let mapper = match v_type.transform(oracle) {
-            Some(item) => format!("?.mapNotNull({item})", item = item),
-            _ => "".into(),
-        };
-
-        format!("{}{}", getter, mapper)
-    }
-
-    /// Accepts two runtime expressions and returns a runtime experession to combine. If the `default` is of type `T`,
-    /// the `override` is of type `T?`.
-    fn with_fallback(
+    fn value_getter(
         &self,
-        _oracle: &dyn CodeOracle,
-        overrides: &dyn Display,
-        default: &dyn Display,
+        oracle: &dyn CodeOracle,
+        vars: &dyn Display,
+        prop: &dyn Display,
     ) -> String {
+        let vtype = oracle.find(&self.inner).variables_type(oracle);
         format!(
-            "{overrides} ?: {default}",
-            overrides = overrides,
-            default = default
+            "{vars}?.get{vt}List(\"{prop}\")",
+            vars = vars,
+            vt = vtype,
+            prop = prop
         )
     }
 
+    fn value_mapper(&self, oracle: &dyn CodeOracle) -> Option<String> {
+        let transform = oracle.find(&self.inner).create_transform(oracle)?;
+        Some(if transform.starts_with('{') {
+            format!("mapNotNull {}", transform)
+        } else {
+            format!("mapNotNull({})", transform)
+        })
+    }
+
+    fn value_merger(&self, _oracle: &dyn CodeOracle, _default: &dyn Display) -> Option<String> {
+        // We never merge lists.
+        None
+    }
+
     /// The name of the type as it's represented in the `Variables` object.
     /// The string return may be used to combine with an indentifier, e.g. a `Variables` method name.
     fn variables_type(&self, _oracle: &dyn CodeOracle) -> VariablesType {
-        unimplemented!("Nesting lists in to lists and maps are not supported")
+        unimplemented!("Because lists do not merge very well, maps of lists or lists of lists are not supported")
```

Comment:
Lists don't merge well (we just replace them), but I don't think this is the reason why we won't support `Map<String, List<String>`.

I think it's because we don't yet have a `variables?.getListList()` or `variables.asStringList` yet. If we have a need for it, then we should support it, but what we're doing is already likely more than we'll ever need.